### PR TITLE
#28 add member management page

### LINKED ISSUE #28: Client: Add member management page
* Manually create members
* Mark member as resigned
* Change status of members

### PR DESCRIPTION
**Pull request by qnewald**
Adds a new page that allows admins to change the status of current members. At this time it is not possible to create new members, because of a dependency on another issue.

Closes #28

### CODE REVIEW

**Comment by qnewald** [comment]
Closes #28

File: client/src/ts/members/MemberManagement.tsx [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+      })
+      .then((res) => {
+        if (res.status === 200) {
+          if (mounted) {
+            setMembers(res.data);
+          }
+        }
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (jbt-it) [inline client/src/ts/members/MemberManagement.tsx:207]: no console log

File: client/src/ts/members/MemberManagement.tsx [hunk 2]
```diff
@@ -0,0 +1,698 @@
...
+    }
+  };
+
+  /**
+   * Changes the status of the member
+   */
+  const changeMemberStatus = (id: number, status: string) => {
+    // TODO: prompt user to confirm status change
+
+    let payload = {
```
Comment (jbt-it) [inline client/src/ts/members/MemberManagement.tsx:376]: use const

File: client/src/ts/members/MemberManagement.tsx [hunk 3]
```diff
@@ -0,0 +1,698 @@
...
+        if (res.status === 200) {
+          getMembers();
+          setSuccessOpen(successOpen + 1);
+          setSuccessOpen(0);
+        } else {
+          setErrorOpen(errorOpen + 1);
+          setErrorOpen(0);
+        }
+      })
+      .catch((error) => {});
```
Comment (jbt-it) [inline client/src/ts/members/MemberManagement.tsx:394]: Empty catch

**Review by jbt-it — CHANGES_REQUESTED** [review]
- Add a way to navigate to the subpage
- give options for the Trainee Generation
- add the "Trainee seit field"

**Comment by arisb1470** [comment]
Looks good! There are two issues that I found: 
When trying to change the status to "ausgetretenes Mitglied, following error message pops up: "Request failed with status code 500"
Secondly, trying to search for members doesn't appear to be working. When entering any symbol into the searchbar, the site goes blank.

**Pull request merged**